.github/workflows/build.yml

```diff
@@ -0,0 +1,107 @@
+# Terraform Provider testing workflow using different terraform versions
+# and clouds.
+
+name: Build
+
+# This GitHub action runs your tests for each pull request.
+on:
+  pull_request:
+    paths-ignore:
+      - "README.md"
+      - "Makefile"
+      - "project-docs/**"
+  push:
+    branches:
+      - "main"
+    paths-ignore:
+      - "README.md"
+      - "Makefile"
+      - "project-docs/**"
+
+
+# Testing only needs permissions to read the repository contents.
+permissions:
+  contents: read
+
+jobs:
+  # Ensure project builds before running testing matrix
+  build:
+    name: Build
+    needs:
+      - generate
```

Comment:
The new dependencies are bit funky here. So integration test currently requires `generate` and `lint`, and they in turn used to require the `build`. But now it looks like build requires `generate`, which sort of makes sense, but looks like this breaks the dependency. For example the integration test would still run if the lint, or format has failed.

My suggestion would be: 

- `lint` needs `format` and `copyright-check`
- `build` needs `generate` and `lint`
- `test` needs `build`

Reply:
Each of these items are independent, no dependencies on another.

Reply:
Well there was a dependency between `test`, `generate` and `lint` that this change breaks. Question is do we want them to be separate? I don't think having the same error message in multiple action outputs is very nice. However, not a big deal to me. Check out [this SF answer](https://stackoverflow.com/a/64733705) for inter-workflow dependency if you wanna make them depend on each other.